artifactory/system.go

```diff
@@ -33,6 +40,1112 @@ func (v Versions) String() string {
+}
+
+// MarshalYAML implements the Marshaller interface.
+func (s SystemMessageConfig) MarshalYAML() (interface{}, error) {
+	if s.Reset != nil && *s.Reset {
+		return nil, nil
+	}
+	return s, nil
+}
+
+// FolderDownloadConfig represents Folder Download settings in Artifactory General Configuration.
+//
+// Docs: https://www.jfrog.com/confluence/display/RTF/YAML+Configuration+File#YAMLConfigurationFile-General(General,PropertySets,Proxy,Mail)
+type FolderDownloadConfig struct {
+	Enabled               *bool `yaml:"enabled,omitempty" xml:"enabled,omitempty"`
+	MaxConcurrentRequests *int  `yaml:"maxConcurrentRequests,omitempty" xml:"maxConcurrentRequests,omitempty"`
+	MaxDownloadSizeMb     *int  `yaml:"maxDownloadSizeMb,omitempty" xml:"maxDownloadSizeMb,omitempty"`
+	MaxFiles              *int  `yaml:"maxFiles,omitempty" xml:"maxFiles,omitempty"`
+	EnabledForAnonymous   *bool `yaml:"enabledForAnonymous,omitempty" xml:"enabledForAnonymous,omitempty"`
+	Reset                 *bool `yaml:"-" xml:"-"`
+}
+
+// MarshalYAML implements the Marshaller interface.
+func (f FolderDownloadConfig) MarshalYAML() (interface{}, error) {
+	if f.Reset != nil && *f.Reset {
+		return nil, nil
+	}
+	return f, nil
+}
+
+// TrashcanConfig represents Trash Can settings in Artifactory General Configuration.
+//
+// Docs: https://www.jfrog.com/confluence/display/RTF/YAML+Configuration+File#YAMLConfigurationFile-General(General,PropertySets,Proxy,Mail)
+type TrashcanConfig struct {
+	Enabled             *bool `yaml:"enabled,omitempty" xml:"enabled,omitempty"`
+	RetentionPeriodDays *int  `yaml:"retentionPeriodDays,omitempty" xml:"retentionPeriodDays,omitempty"`
+	AllowPermDeletes    *bool `yaml:"allowPermDeletes,omitempty" xml:"allowPermDeletes,omitempty"`
+	Reset               *bool `yaml:"-" xml:"-"`
+}
+
+// MarshalYAML implements the Marshaller interface.
+func (t TrashcanConfig) MarshalYAML() (interface{}, error) {
+	if t.Reset != nil && *t.Reset {
+		return nil, nil
+	}
+	return t, nil
+}
+
+// ReplicationsConfig represents Global Replication Blocking
+// settings in Artifactory General Configuration.
+//
+// Docs: https://www.jfrog.com/confluence/display/RTF/YAML+Configuration+File#YAMLConfigurationFile-General(General,PropertySets,Proxy,Mail)
+type ReplicationsConfig struct {
+	BlockPullReplications *bool `yaml:"blockPullReplications,omitempty" xml:"blockPullReplications,omitempty"`
+	BlockPushReplications *bool `yaml:"blockPushReplications,omitempty" xml:"blockPushReplications,omitempty"`
+	Reset                 *bool `yaml:"-" xml:"-"`
+}
+
+// MarshalYAML implements the Marshaller interface.
+func (r ReplicationsConfig) MarshalYAML() (interface{}, error) {
+	if r.Reset != nil && *r.Reset {
+		return nil, nil
+	}
+	return r, nil
+}
+
+// BintrayApplication represents Bintray Oauth applications configuration.
+// This is undocumented in YAML Configuration File.
+//
+type BintrayApplication struct {
+	Key          *string `yaml:"-" xml:"key,omitempty"`
```

Comment:
I noticed we're not allowing this field to be serialized in YAML due to the `-` tag.

This peaked my curiosity so was this intentional?

Reply:
Similar to [comment](https://github.com/target/go-arty/pull/57#discussion_r436119180).